### PR TITLE
DCS-909 implemented remaining edits required in PR#499 as we…

### PR DESCRIPTION
**Pull request by louisenorris**
…ll as request removal navigation fix

### CODE REVIEW

**Comment by louisenorris** [comment]
(Integration tests still to update in next PR)

File: server/config/incident.js [hunk 1]
```diff
@@ -23,6 +25,15 @@ const paths = {
...
   removalRequested: () => '/removal-requested',
+  viewRemovalRequest: (reportId, statementId) =>
+    `/coordinator/report/${reportId}/statement/${statementId}/view-removal-request`,
+  staffMemberNotRemoved: (reportId, statementId) =>
+    `/coordinator/report/${reportId}/statement/${statementId}/staff-member-not-removed`,
+  confirmStatementDelete: (reportId, statementId, removalRequest) =>
+    `/coordinator/report/${reportId}/statement/${statementId}/confirm-delete${
+      removalRequest ? `?removalRequest=${removalRequest}` : ''
+    }`,
+  confirmReportDelete: reportId => `/coordinator/report/${reportId}/confirm-delete`,
```
Comment (andrewrlee) [inline server/config/incident.js:36]: I've been sort of misusing this - this is supposed to be a "create report" specific file.. we could probably do with either extracting the routes out into somewhere else or breaking down based on application section ("createReport", "manageReports" etc...)

This is obviously completely fine as it is though.. that would be done separate PR

File: server/routes/maintainingReports/coordinator.test.ts [hunk 2]
```diff
@@ -497,7 +498,7 @@ describe('coordinator', () => {
           email: 'bob@gmail.com',
         })
         await request(app)
-          .get('/coordinator/report/123/statement/2/staff-member-not-removed')
+          .get(paths.staffMemberNotRemoved(123, 2))
```
Comment (andrewrlee) [inline server/routes/maintainingReports/coordinator.test.ts:501]: great to have updated all these 👍

File: server/routes/maintainingReports/coordinator.ts [hunk 3]
```diff
@@ -167,30 +167,34 @@ export default class CoordinatorRoutes {
...
+        ? res.redirect(paths.confirmStatementDelete(reportId, statementId, true))
+        : res.redirect(paths.confirmStatementDelete(reportId, statementId))
     }
 
     if (confirm === 'yes') {
       await this.involvedStaffService.removeInvolvedStaff(reportId, parseInt(statementId, 10))
     }
 
-    return res.redirect(`/${reportId}/view-report`)
+    const location = removalRequest ? `/${reportId}/view-statements` : `/${reportId}/view-report`
```
Comment (andrewrlee) [inline server/routes/maintainingReports/coordinator.ts:197]: this could use paths too

File: server/routes/maintainingReports/coordinator.ts [hunk 4]
```diff
@@ -167,30 +167,34 @@ export default class CoordinatorRoutes {
...
     const { statementId } = req.params
-    const { confirm } = req.body
+    const { confirm, removalRequest } = req.body
 
     if (!confirm) {
       req.flash('errors', [{ href: '#confirm', text: 'Select yes if you want to delete this statement' }])
-      return res.redirect(`/coordinator/report/${reportId}/statement/${statementId}/confirm-delete`)
+      return removalRequest
+        ? res.redirect(paths.confirmStatementDelete(reportId, statementId, true))
+        : res.redirect(paths.confirmStatementDelete(reportId, statementId))
```
Comment (andrewrlee) [inline server/routes/maintainingReports/coordinator.ts:190]: we probably should have an extra test to cover this

File: server/routes/maintainingReports/coordinator.ts (hunk 3 above)
Comment (andrewrlee) [reply server/routes/maintainingReports/coordinator.ts:197]: we probably should have an extra test to cover this

File: server/views/pages/coordinator/staff-member-not-removed.html [hunk 5]
```diff
@@ -32,7 +32,7 @@ <h2 class="govuk-heading-l">
           </div>
       </dl>
 
-      <p><a href={{data.returnToIncidentLink}}>Return to use of force incident</a></p>
+      <p><a href="/{{ data.reportId }}/view-statements">Return to use of force incident</a></p>
```
Comment (andrewrlee) [inline server/views/pages/coordinator/staff-member-not-removed.html:35]: not related to this PR, I had a thought we could make the links object as part of the `app.locals` (or `res.locals`) and then use it in templates too: `href="{{ links.viewStatement(data.reportId) }}` or the like

File: server/routes/maintainingReports/coordinator.ts (hunk 4 above)
Comment (louisenorris) [reply server/routes/maintainingReports/coordinator.ts:190]: Done

File: server/routes/maintainingReports/coordinator.test.ts [hunk 6]
```diff
@@ -339,13 +340,26 @@ describe('coordinator', () => {
   })
 
   describe('Delete statement', () => {
-    it('not confirming deletion triggers validation', async () => {
+    it('Validation error redirects back to current page', async () => {
+      userSupplier.mockReturnValue(coordinatorUser)
+
+      await request(app)
+        .post('/coordinator/report/123/statement/2/delete')
```
Comment (andrewrlee) [inline server/routes/maintainingReports/coordinator.test.ts:347]: meh.. these could use paths

**Review by andrewrlee — APPROVED** [review]
LGTM

File: server/routes/maintainingReports/coordinator.ts (hunk 3 above)
Comment (louisenorris) [reply server/routes/maintainingReports/coordinator.ts]: Done
